### PR TITLE
[not for merge] Update to fastparse 2 (#510)

### PR DESCRIPTION
**Pull request by raboof**
Refs #510.

I had to disable the 'native' build targets for now since fastparse 2 has not been released for Scala Native yet, see https://github.com/lihaoyi/fastparse/issues/215.

### CODE REVIEW

File: scalapb-runtime/shared/src/main/scala/scalapb/textformat/ProtoAsciiParser.scala [hunk 1]
```diff
@@ -1,48 +1,37 @@
...
-  val MessageValue: P[TMessage] = P(
+  def MessageValue[_: P]: P[TMessage] = P(
     Index ~ "{" ~/ KeyValue.rep ~/ "}" |
       Index ~ "<" ~/ KeyValue.rep ~/ ">"
   ).map(TMessage.tupled)
 
-  val ValueArray: P[TValue] =
-    P((Index ~ "[" ~/ (PrimitiveValue | MessageValue).rep(0, ",".~/) ~/ "]")).map(TArray.tupled)
+  def ValueArray[_: P]: P[TValue] =
+    P((Index ~ "[" ~/ (PrimitiveValue | MessageValue).rep(0, ",") ~/ "]")).map(TArray.tupled)
```
Comment (raboof) [inline scalapb-runtime/shared/src/main/scala/scalapb/textformat/ProtoAsciiParser.scala]: I'm not sure the `.~/` was meaningful here?

File: scalapb-runtime/shared/src/main/scala/scalapb/textformat/ProtoAsciiParser.scala [hunk 2]
```diff
@@ -1,48 +1,37 @@
 package scalapb.textformat
 
-import fastparse.WhitespaceApi
+import fastparse._
+import fastparse.ScriptWhitespace._
```
Comment (raboof) [inline scalapb-runtime/shared/src/main/scala/scalapb/textformat/ProtoAsciiParser.scala]: I'm not entirely sure this correctly corresponds to `Basics.whiteSpace`

**Comment by SethTisue** [comment]
this appears to have been merged in some form...?

**Comment by raboof** [comment]
Looks like :+1:!

**Comment by thesamet** [comment]
Yes, thanks again for this contribution.

**Pull request closed without merging**